### PR TITLE
Fix bug 1226759. A click on placeable replaces selected text.

### PR DESCRIPTION
**Pull request by jotes**
Hi @mathjazz! Could you review?

### CODE REVIEW

File: pontoon/base/static/js/translate.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 /* Extend public object */
 var Pontoon = (function (my) {
   return $.extend(true, my, {
-
```
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Let's keep this. For consistency (we have a blank line at the bottom, too).

File: pontoon/base/static/js/translate.js [hunk 2]
```diff
@@ -657,14 +656,18 @@ var Pontoon = (function (my) {
       $("#original").on("click", ".placeable", function (e) {
         e.stopPropagation();
         e.preventDefault();
+        $('#translation').focus();
```
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: We don't need this, we already focus in line 669.

File: pontoon/base/static/js/translate.js [hunk 3]
```diff
@@ -657,14 +656,18 @@ var Pontoon = (function (my) {
...
 
         var textarea = $('#translation'),
             pos = textarea[0].selectionStart,
```
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Remove this line.

File: pontoon/base/static/js/translate.js [hunk 4]
```diff
@@ -657,14 +656,18 @@ var Pontoon = (function (my) {
...
             placeable = $(this).text(),
             before = textarea.val(),
-            after = before.substring(0, pos) + placeable + before.substring(pos);
+            after = before.substring(0, pos) + placeable + before.substring(pos),
```
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Remove this line (you override it later).

File: pontoon/base/static/js/translate.js [hunk 5]
```diff
@@ -657,14 +656,18 @@ var Pontoon = (function (my) {
...
+            cursorPos = textarea.prop('selectionStart') + placeable.length,
+            after = before.substring(0, textarea.prop('selectionStart')) + placeable + before.substring(textarea.prop('selectionEnd'))
```
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Add semicolon at the end.

File: pontoon/base/static/js/translate.js [hunk 6]
```diff
@@ -657,14 +656,18 @@ var Pontoon = (function (my) {
...
 
         textarea.val(after).focus();
+        setCaretToPos(textarea[0], cursorPos);
```
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Why not simply replace this with `input.setSelectionRange(pos, pos);`?

**Comment by mathjazz** [comment]
It works great, thanks!

The code could be simplified though.

File: pontoon/base/static/js/translate.js (hunk 2 above)
Comment (jotes) [inline pontoon/base/static/js/translate.js]: For some reason, i couldn't get selectionStart/selectionEnd properties without this line. However, it works fine by now. Thanks!

**Comment by jotes** [comment]
@mathjazz could you review again?

**Comment by mathjazz** [comment]
Great job! r+

**Pull request merged**